lib/fizzy/execute.cpp

```diff
@@ -508,7 +508,8 @@ std::unique_ptr<Instance> instantiate(Module module,
     datasec_offsets.reserve(module.datasec.size());
     for (const auto& data : module.datasec)
     {
-        const uint64_t offset = eval_constant_expression(data.offset, imported_globals, globals);
+        const uint64_t offset =
+            eval_constant_expression(data.offset, imported_globals, globals).i64;
```

Comment:
Aren't these two validated to be `uin32_t`? And if so, the check below doesn't need to operate on 128-bit for overflow checks.

Reply:
It's validated to be 32 bit, and this, together with validation of `data.init` vector length to be 32 bit, guarantees that `offset + data.init.size()` won't overflow 64 bit.

If we do here `as<uint32_t>()`, the check below will implicitly cast it back to 64-bit anyway, so I'm not sure it's worth it.

Reply:
Perhaps adding a note that these are actually 32-bit could be useful?

Reply:
Added comment here and for elem section.